Ensure Static Graph restore is available in dotnet.exe

This basically means ingesting https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Console/NuGet.Build.Tasks.Console.csproj into the SDK. 

We should discuss with the .NET Core team what branches they want the changes in. 

fyi @jeffkl

> I believe the package is NuGet.Build.Tasks.Console and it works because it's just adding its stuff to any\contentFiles like the other packages.

CSC : error CS2015: 'F:\workspace\_work\1\s\.packages\\nuget.build.tasks.console\5.6.0-preview.1.6478\contentFiles\any\netcoreapp2.1\NuGet.Build.Tasks.Console.dll' is a binary file instead of a text file [F:\workspace\_work\1\s\src\Layout\redist\redist.csproj]
##[error]CSC(0,0): error CS2015: 'F:\workspace\_work\1\s\.packages\\nuget.build.tasks.console\5.6.0-preview.1.6478\contentFiles\any\netcoreapp2.1\NuGet.Build.Tasks.Console.dll' is a binary file instead of a text file

from https://dev.azure.com/dnceng/public/_build/results?buildId=550640.

@jeffkl putting a binary into the contentFiles directory isn't allowed nor recommended. We are usually putting our executable binaries into the tools directory as recommended in the docs. Any reason why you chose contentFiles?

--- a/src/NuGet.Core/NuGet.Build.Tasks.Console/NuGet.Build.Tasks.Console.csproj
+++ b/src/NuGet.Core/NuGet.Build.Tasks.Console/NuGet.Build.Tasks.Console.csproj
@@ -15,7 +15,6 @@
     <XPLATProject>true</XPLATProject>
     <EnableDefaultNoneItems>false</EnableDefaultNoneItems>
     <ServerGarbageCollection>true</ServerGarbageCollection>
-    <BuildOutputTargetFolder>contentFiles\any</BuildOutputTargetFolder>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
   </PropertyGroup>
 
@@ -36,6 +35,14 @@
 
   <ItemGroup>
     <None Include="App.config" Condition=" '$(TargetFramework)' == '$(NETFXTargetFramework)' " />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="..\NuGet.Build.Tasks\NuGet.RestoreEx.targets">
+      <Link>%(Filename)%(Extension)</Link>
+      <PackagePath>runtimes\any\native</PackagePath>
+      <Pack>true</Pack>
+    </Content>
   </ItemGroup>
 
   <ItemGroup Condition="$(DefineConstants.Contains(SIGNED_BUILD))">

--- a/src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.csproj
+++ b/src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.csproj
@@ -1,4 +1,4 @@
-﻿<Project>
+<Project>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'README.md'))\build\common.props" />
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
@@ -21,8 +21,7 @@
   <ItemGroup>
     <Content Include="NuGet.RestoreEx.targets">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-      <PackagePath>runtimes\any\native</PackagePath>
-      <Pack>true</Pack>
+      <Pack>false</Pack>
     </Content>
     <Content Include="NuGet.targets">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>

